controllers/managedfusion_controller.go

```diff
@@ -297,26 +298,39 @@ func (r *ManagedFusionReconciler) initReconciler(ctx context.Context, req ctrl.R
 
 func (r *ManagedFusionReconciler) reconcilePhases() (reconcile.Result, error) {
 	if !r.managedFusionSecret.DeletionTimestamp.IsZero() {
-		if r.verifyOfferringsDoNotExist() {
-			r.Log.Info(fmt.Sprintf("removing finalizer from %s secret", managedFusionSecretName))
-			r.managedFusionSecret.SetFinalizers(utils.Remove(r.managedFusionSecret.GetFinalizers(), managedFusionFinalizer))
-			if err := r.Client.Update(r.ctx, r.managedFusionSecret); err != nil {
-				return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from %s secret: %v", managedFusionSecretName, err)
+		r.Log.Info("Managed Fusion configuration Secret is marked for deletion, checking for ManagedFusionOfferings CRs on the cluster")
+		if found, err := r.checkForManagedOfferings(); err != nil {
+			return ctrl.Result{}, err
+		} else if found {
+			r.Log.Info("ManagedFusionOffering CRs found, issuing delete")
+			offering := v1alpha1.ManagedFusionOffering{}
+			if err := r.deleteAllOf(&offering); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to issue delete for one or more ManagedFusionOffering CRs, %v", err)
+			} else {
+				r.Log.Info("ManagedFusionOffering CRs delete issued, requeuing the event")
+				return ctrl.Result{RequeueAfter: 10}, nil
+			}
+		} else {
+			r.Log.Info("No offering found, removing finalizer from %s secret", managedFusionSecretName)
```

Comment:
Use `fmt.Sprintf()` or pass key and value to `r.Log.Info()`
```suggestion
			r.Log.Info(fmt.Sprintf("No offering found, removing finalizer from %s secret", managedFusionSecretName))
```

Reply:
done